src/web/components/StickyBottomBanner/StickyBottomBanner.tsx

```diff
@@ -13,15 +13,15 @@ type Props = {
     asyncCountryCode?: Promise<string>;
     CAPI: CAPIBrowserType;
     asyncBrazeUuid?: Promise<null | string>;
-    isDigitalSubscriber?: boolean;
+    shouldHideSupportMessaging?: boolean;
```

Comment:
I wonder if we could encapsulate our logic a bit better here by naming this something that *we* care about, rather than having the name be based on the (current) source for the data?

The actual `ts` logic is where we do our business logic to decide how to render the component, and the `tsx` component only needs to be given that information. This feels a bit like leaking the implementation details into the template. Is the actual concept we're trying to convey here something along the lines of "this impression is eligible for Braze Web Messages", and if so would a name along those lines make it:
- easier to understand these template files in isolation
- easier to refactor the "can show" logic to consider other cases in the future

Reply:
Hey yeah this is a great point, thanks for raising. Rhys and I were discussing something similar on Friday. I agree that at some point we need to map the source of the data `shouldHideSupportMessaging` to the business logic we care about. This component, `StickyBottomBanner`, is the orchestrator of which banner actually gets shown and part of its responsibility is gathering and passing on the inputs which allow each of the banner `canShow` functions to make a decision. So to me it seems like the `canShow` of the relevant banner is the place where the mapping should take place. I think that's consistent with how the reader revenue `canShow` works for example. Happy to chat about this!

Reply:
I've added a comment to the preChecks in `BrazeBanner.tsx` which hopefully makes things clearer to future developers (including ourselves!)